org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/extensions/XMLExtensionsRegistry.java

```diff
@@ -382,4 +390,8 @@ public void setNotificationService(IXMLNotificationService notificationService)
 		this.notificationService = notificationService;
 	}
 
+	public XMLLanguageServer getServer() {

```

Comment:
It seems this getter is never used?

Reply:
It is used in the Spring XML server extension to:
- register/unregister command
- handle the registered command
- access Doc Service to call `triggerValidationFor(doc)`

I'll think how wrap this inside some API not to call the server directly. If you have an good idea how to wrap this into nice API please let me know here :-) Otherwise, I'll provide something soon.

Reply:
> * register/unregister command
> * handle the registered command

Do that in the XMLExtensionRegistry (like completion participant)

> * access Doc Service to call triggerValidationFor(doc)

I suggest to provide a new IXMLValidationService interface (like we have for IXMLNotificationService and IDocumentProvider). The XMLTextDocumentService implement IXMLValidationService and the XMLLanguageServer set this instance with XMLExtensionregsity#setValidationService